test/unit/tasks/download-assets.test.js

```diff
@@ -84,8 +120,47 @@ test('Downloads assets and properly counts failed attempts', () => {
       expect(ctx.assetDownloads).toEqual({
         successCount: 2,
         warningCount: 1,
-        errorCount: 1
+        errorCount: 2
+      })
+      expect(output.mock.calls).toHaveLength(5)
+    })
+})
+
+test('it doesn\'t use fileStack url as fallback for the file url and throws a warning output', () => {
+  const task = downloadAssets({
+    exportDir: tmpDirectory
+  })
+  const output = jest.fn()
+  const taskProxy = new Proxy({}, {
+    set: (obj, prop, value) => {
+      if (prop === 'output') {
+        output(value)
+        return value
+      }
+      throw new Error(`It should not access task property ${prop} (value: ${value})`)
+    }
+  })
+  const ctx = {
+    data: {
+      assets: [
+        ...getAssets({ existing: 2, missingUrl: 1 })
+      ]
+    }
+  }
+
+  return task(ctx, taskProxy)
+    .then(() => {
+      expect(ctx.assetDownloads).toEqual({
+        successCount: 4,
+        warningCount: 2,
+        errorCount: 0
       })
-      expect(output.mock.calls).toHaveLength(4)
+      expect(output.mock.calls).toHaveLength(6)
+
+      const [missingUrlOut1] = output.mock.calls[0]
```

Comment:
If I understand correctly, this assumes that the first two calls to `task.output` are the ones for the assets with a missing `url` property. This makes sense but if someone changes the order of the array in the `getAssets` function this test will fail. 

Can we maybe map over `output.mock.calls` using `array.filter` and then check that it finds two matches?